### PR TITLE
SSR Support #292

### PR DESCRIPTION
**Pull request by shimarulin**
Changes:

- String.prototype.toLowerCase was replaced to [lodash.tolower](https://www.npmjs.com/package/lodash.tolower)
- Type checking for `window`, `document` and `navigator` was added
- Browser only types of object were removed

### CODE REVIEW

File: src/internal/abstractButton.js [hunk 1]
```diff
@@ -179,7 +179,7 @@ export default {
       }
     },
     getTagName () {
-      const isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') !== -1
+      const isFirefox = typeof navigator !== 'undefined' && navigator.userAgent.toLowerCase().indexOf('firefox') !== -1
```
Comment (mystadio) [inline src/internal/abstractButton.js]: @shimarulin Shouldn't it be `toLower(navigator.userAgent)` instead?

Comment (shimarulin) [inline src/internal/abstractButton.js]: `lodash.toLower` used only on server-side. How you can see, `navigator.userAgent.toLowerCase()` will called only if `navigator` is defined, that works on client-side only. Currently [all browsers have String.prototype.toLowerCase method](http://caniuse.com/#search=tolowercase). Therefore, there is no need to replace

P.S.: I don't understand why Node.js doesn't have an implementation of such a simple method, but it's fact

**Comment by juzhiyuan** [comment]
then when can I make use of the new feature please?

**Comment by cnole** [comment]
Any plans for a new version? I am using 2.0.2 currently und ssr support is not yet there.

**Comment by n1ru4l** [comment]
+1

**Comment by jimliang** [comment]
one more
```javascript
ReferenceError: window is not defined
```
in `src/internal/scroll.js`

```javascript
props: {
    scroller: {
      default () {
        return window
      }
    }
  }
```

**Comment by myronliu347** [comment]
ok， I find

**Comment by myronliu347** [comment]
Sorry, I will  fix

**Pull request merged**